### DISCUSSION TITLE
Installation error on M1 Mac

### DISCUSSION THREAD

**Discussion by kimiaGF** [SINGROUP/dscribe#78] [Q&A] [answered]
I'm following the installation instructions for building from source, but after "pip install ."  the command runs until it gets to building numpy, then outputs the following error:
 ERROR: Failed building wheel for numpy
Failed to build numpy
ERROR: Could not build wheels for numpy which use PEP 517 and cannot be installed directly

pip version: 21.3.1
python version: 3.9.7

running command on virtual environment

**Comment by lauri-codes** [comment, marked as answer]
Hi @kimiaGF!

This seems like a numpy installation issue. With a quick google search it looks like it may be some kind of a compatibility issue with the M1 chip and the standard numpy package. I would suggest trying e.g. [the instructions in this stack overflow discussion](https://stackoverflow.com/questions/65336789/numpy-build-fail-in-m1-big-sur-11-1).

**Reply by kimiaGF** [reply]
Hi @lauri-codes! Thanks for taking the time to help me out with this, it's really blocking any progress in my research.

What's weird about this is that I have installed and used numpy before on my Laptop. I'm also able to individually download numpy, llvmlite, and numba through conda but when "pip install"-ing it gives me the same error that the wheels for numpy, llvmlite, and numba could not be installed. I also tried "conda install dscribe" and I get the following error:

Solving environment: failed with initial frozen solve. Retrying with flexible solve.

```
PackagesNotFoundError: The following packages are not available from current channels:

  - dscribe

Current channels:

  - https://repo.anaconda.com/pkgs/main/osx-arm64
  - https://repo.anaconda.com/pkgs/main/noarch
  - https://repo.anaconda.com/pkgs/r/osx-arm64
  - https://repo.anaconda.com/pkgs/r/noarch
  - https://conda.anaconda.org/conda-forge/osx-arm64
  - https://conda.anaconda.org/conda-forge/noarch

To search for alternate channels that may provide the conda package you're
looking for, navigate to

    https://anaconda.org

and use the search bar at the top of the page.
```
when I go to anaconda.org and search for dscribe I see that it's there but conda can't seem to find it. Do you know what could be causing this? I'm also attaching the full error log that outputs when I use "pip install ." to build from source. The same error is given even if I specify `conda-forge`
[ERROR: Command errored out with exit status 1:.pdf](https://github.com/SINGROUP/dscribe/files/7831010/ERROR.Command.errored.out.with.exit.status.1.pdf)

**Reply by kimiaGF** [reply]
**Found a solution!**

Something was wrong with my conda so uninstalling it and reinstalling anaconda worked. I think I may have deleted some miniforge files that broke my conda, but regardless the following code solved my issue and I was able to install dscribe on my M1 MacBook Air:

`pip uninstall conda`

install anaconda again (NOT miniconda!)

`conda install dscribe`
